yarn-project/circuits.js/src/structs/kernel/historic_block_data.ts

```diff
@@ -78,7 +83,8 @@ export class HistoricBlockData {
   }
 
   toString() {
-    return this.toBuffer().toString();
+    // originally this was encoding as utf-8 (the default). This caused problems decoding back the data.
+    return this.toBuffer().toString(STRING_ENCODING);
```

Comment:
Originally this method didn't specify an encoding so NodeJS used UTF-8 as the default. The actual data in the buffer doesn't necessarily include valid UTF-8 bytes sequences so this would cause issues when deserliasing from a string. I have changed the encoding to `hex` now and added a test.